registry/client/repository.go

```diff
@@ -604,6 +607,25 @@ func (bs *blobs) Resume(ctx context.Context, id string) (distribution.BlobWriter
 	panic("not implemented")
 }
 
+func (bs *blobs) Mount(ctx context.Context, sourceRepo distribution.Repository, dgst digest.Digest) (distribution.Descriptor, error) {
```

Comment:
@stevvooe: @BrianBland's engine side code implements a minimal `Repository` implementation that just returns the name, so it's not that burdensome. But it would be even simpler to just take the name, if that doesn't have any downsides.